packages/lit-dev-content/site/tutorials/content/working-with-lists/02.md

```diff
@@ -0,0 +1,14 @@
+We've see how we can use a `for ... of` loop to populate an array of templates
+to render. However, Lit also provides a built-in
+[`map`](https://lit.dev/docs/templates/directives/#map) directive for a
+more declarative way of writing templates.
```

Comment:
I'd drill into this a bit more. Explain why the map directive exists it's declarative, provides something similar to the .map interface to generic iterables, and makes your code cleaner to read.

Reply:
Thanks Elliott! The fact that `map` works on the `Iterable` type is quite powerful :P

[A quick playground experiment](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy5qcyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7bWFwfSBmcm9tICdsaXQvZGlyZWN0aXZlcy9tYXAuanMnO1xuXG5mdW5jdGlvbiogd2lsZEl0ZXJhYmxlKCkge1xuICBmb3IgKGxldCBpID0gMDsgaSA8IDQ7IGkrKykge1xuICAgIHlpZWxkIGk7XG4gIH1cbn1cblxuZXhwb3J0IGNsYXNzIFNpbXBsZUdyZWV0aW5nIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGA8dWw-JHttYXAod2lsZEl0ZXJhYmxlKCksICh2KSA9PiBodG1sYDxsaT4ke3Z9PC9saT5gKX08L3VsPmA7XG4gIH1cbn1cbmN1c3RvbUVsZW1lbnRzLmRlZmluZSgnc2ltcGxlLWdyZWV0aW5nJywgU2ltcGxlR3JlZXRpbmcpO1xuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxzaW1wbGUtZ3JlZXRpbmcgbmFtZT1cIldvcmxkXCI-PC9zaW1wbGUtZ3JlZXRpbmc-XG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMi4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4xLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeMy4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjIuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d)